bin/tools.py

```diff
@@ -1043,13 +1044,16 @@ def read_personal_config():
     args = {}
 
     if is_windows():
-        home_config = Path(os.getenv('AppData'))
+        app_data = os.getenv('AppData')
+        if not app_data:
+            fatal("%AppData% does not exist")
+        home_config = Path(app_data)
     else:
-        home_config = (
-            Path(os.getenv('XDG_CONFIG_HOME'))
-            if os.getenv('XDG_CONFIG_HOME')
-            else Path(os.getenv('HOME')) / '.config'
-        )
+        home = os.getenv('HOME')
+        if not home:
+            fatal("$HOME does not exist")
```

Comment:
Return empty `args` instead of `fatal`?

Reply:
Yeah fair, done :slightly_smiling_face: I mean, people's systems are probably utterly broken (or misusing `sudo`) if the `$HOME` or `%AppData%` variables are missing, but we don't need to judge them by throwing a fatal error :stuck_out_tongue: